### ISSUE TITLE
Gun refactoring (and making the Lee-Enfield a little less pointless)

### ISSUE THREAD

**Issue by rakiru** [iamgreaser/iceball#117]
Add spread multiplier/addition options for different players states (jumping, crouching, etc.) and/or speed or something, and scoped or not.

Remove the sway from the Lee-Enfield when it's not scoped in, but add somewhat insane spread (think Counter Strike AWP)

**Comment by LeComm** [comment]
It also looks like if its clip only contains one bullet and the player has to reload after every shot, rather than just having to pull the bolt after every shot.

**Comment by rakiru** [comment]
I'm not sure what you're trying to say. Yes, that's how it works right now.

**Comment by LeComm** [comment]
Pulling a bolt is much faster than reloading. It would make more sense to encrease the delay between shots, add an animation and set the clip to 8-10.

**Comment by Dany0** [comment]
adding an animation right now is hard, all animations in the game right now are hacks

we don't exactly have lib_anim.lua lib_tween.lua

**Comment by rakiru** [comment]
An animation framework has been hinted at, but we don't currently have one. Yes, I would prefer a reload animation, but since that isn't currently possible, it's just a regular reload. The reload time is so high due to how powerful the gun works; it's more an issue of balance than trying to be a real lee-enfield. I'd be inclined to rename it to "Sniper" or something. It was created as a test of the weapon API, and doesn't really fit the current theme of "generic guns".

**Comment by Dany0** [comment]
It's not actually too powerful because it's not hard to get headshots with the rifle.

And yes exactly, the reason it exists at all is to have two guns so that we can test the weapon API.

Suggestions new weapons are welcome.

**Comment by LeComm** [comment]
Yes I see, but now is probably a good moment to make them properly.

**Comment by rakiru** [comment]
I know, I'm just saying that's why it was created like that. It is easily the most unbalanced and underpowered gun in the game right now, mostly due to it just being a test originally.

Personally, I'd like to make it like the AWP from Counter Strike (oh, I said that already), just to make it a bit more useful, and make it more different than the current rifle.

And yes, new weapon suggestions would be great, as long as they contain a decent idea of what to do, rather than just "make me an akm cuz its my favourite gun".

**Comment by LeComm** [comment]
Completely new weapons or something to replace the current ones?

**Comment by Warp00** [comment]
If I can contribute here, surely the point of new weapons is to suit different play styles/provide variety?

If so, something like a Lee Enfield (.303 Master Race) is a good idea. So is something like an AKM or MP5 (No, not the AoS .75 way ;))

**Comment by rakiru** [comment]
Well, there's no real reason to replace the current ones (except the lee-enfield, since basically everything about it is going to change). New weapons. They'll be implemented as mods, so servers can run only the ones they want anyway.

**Comment by Warp00** [comment]
Is there potential for client mods, or just server mods? Sorry, just trying to find my feet here :). Ofc We can't have players setting weapon values.

**Comment by LeComm** [comment]
@Warp, both.
And imho, the current 4 weapons we have are enough.

**Comment by rakiru** [comment]
@Warp00
All code is sent from the server, so you can do literally anything you want in the sense of gameplay. For example, there is currently a server where you play snake; guns don't even exist there.

@LeComm
There is plenty of space for additional guns, as long as they're not simply re-skins of existing ones. As stated above, the server owner controls what guns are actually available, so even if we had 10 guns, there's no reason you can't still have the more traditional rifle-smg-shotgun combo on a server (those three guns are fairly well balanced in a rock-paper-scissors kind of way right now too, so they work well together).

**Comment by LeComm** [comment]
You think relying so much on server admins is a great idea?

**Comment by rakiru** [comment]
The whole point of Iceball is that it's modable, so of course server admins have the choice. Currently, the default mod list is empty, meaning they have the Rifle and the Lee-Enfield.

**Comment by Chameleonhider** [comment]
Well for concrete weapon ideas... We can take a look at other games (not exactly copy them).
There was a game "Infiltration 2.9", it aimed to be realistic (it was in terms of weapon behavior). So a couple of things in that game: 
(you should look into 1., 3., 4.; other are not as important)
1. Doing sit-ups or moving messes up (shakes) your aim/view a bit so you don't spam crouch, especially when aiming.
2. Holding alt-fire (MOUSE2) uses "control breath" feature, which eliminates sway and makes spread almost zero. Automatic guns are able to snipe too if you wish to conserve ammo, but it gets tiring. Plus second shot will have regular spread, you have to release alt-fire and hold it again.
In IB it could be automatic, after aiming rifle during 5 second delay spread would minimize. You can move while aiming, but spread will need 1-3 seconds to decrease again. Aimed gun spread without that minimization shouldn't be big, but noticeable at fog range.
3. Dry and tactical reloads. Dry reload drops empty magazine. After tactical reload most guns have MagazineCapacity+1 bullets.
4. Weapon fire mode key. Don't think it can only change between Auto/Semi-Auto. If there's a grenade launcher (not RPG) it would control range. 
In a bolt-action rifle it switches between automatic and manual bolt cycling (key to cycle bolt: Reload; you are able to eject live rounds, changes magazine after last shot).
5. Weapon attachment key. Switches between scope and Back-up-ironsights on scoped weapons. Changes ammo type for grenade launcher.
6. Deploying weapon on walls. It was a bind in INF, but it could be automatic in IB. Deploying would occur 1-3 seconds after aiming near (playerheight-1) high wall. Reduces sway and recoil.
7. Weapon collision with wall. This is the reason long sniper rifles, long rifles in general are not used in CQB. Being too close to a wall (facing it) will make you unable to aim your weapon. Hipped weapon moves toward the user, instead of penetrating the wall with its barrel. Player that is already aiming his rifle cannot look towards the wall if rifle doesn't fit in-between.

tl;dr I want to manually cycle the bolt after each shot with my reload key. Not-empty magazine would be reloaded if you hold R for a second, instead of tapping it. You can cycle live rounds too, they are forever lost in the dirt.

**Comment by iamgreaser** [comment]
Suggestions should really be on the subreddit (/r/iceball), not here.
